server/src/main/java/org/elasticsearch/threadpool/ThreadPool.java

```diff
@@ -74,28 +74,50 @@ public class ThreadPool implements ReportingService<ThreadPoolInfo>, Scheduler,
      */
     public static class Names {
         /**
-         * All the tasks that do not relate to the purpose of one of the other thread pools should use this thread pool. Try to pick one of
-         * the other more specific thread pools where possible.
+         * A thread pool with a very high (but finite) maximum size for when there really is no other choice.
+         * <p>
+         * This pool may be used for one-off CPU-bound activities, but the maximum size is so high that it doesn't really work well to do a
+         * lot of CPU-bound work in parallel here. Likewise you can do IO on this pool, but using it for lots of concurrent IO is likely
```

Comment:
Is this another way of saying that CPU-bound work will progress slowly due to a relatively high thread count to available CPUs? You say that it won't work well for a lot of CPU-bound work, but maybe it would be better to spell out why for the reader to avoid guessing?